src/main/java/com/fortanix/sdkms/performance/sampler/KeyGenerationSampler.java

```diff
@@ -79,9 +82,21 @@ public Object execute() throws ApiException {
 
     @Override
     public void teardownTest(JavaSamplerContext context) {
+        this.login(); /* Forcefully re-authenticates thread in case the idle-timeout has kicked
+                         in and expires the session */
+
         for (String keyId : keyIds) {
             try {
                 this.securityObjectsApi.deleteSecurityObject(keyId);
+
+                /* Adding a logging statement that prints the key being deleted, at random intervals
```

Comment:
Its better to print in a non random way to avoid any ambiguous interpretation of the output. We can print a log for every 100 keys deleted per thread.